ui/lib/pki/addon/routes/overview.js

```diff
@@ -11,34 +11,67 @@ export default class PkiOverviewRoute extends Route {
     return this.window || window;
   }
 
-  hasConfig() {
+  async hasConfig() {
     // When the engine is configured, it creates a default issuer.
     // If the issuers list is empty, we know it hasn't been configured
     const endpoint = `${this.win.origin}/v1/${this.secretMountPath.currentPath}/issuers?list=true`;
-    return this.auth
-      .ajax(endpoint, 'GET', {})
-      .then(() => true)
-      .catch(() => false);
+
+    try {
+      await this.auth.ajax(endpoint, 'GET', {});
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async fetchEngine() {
+    const model = await this.store.query('secret-engine', {
+      path: this.secretMountPath.currentPath,
+    });
+    return model.get('firstObject');
+  }
+
+  async fetchAllRoles() {
+    try {
+      return await this.store.query('pki/role', { backend: this.secretMountPath.currentPath });
+    } catch (e) {
+      return e.httpStatus;
+    }
+  }
+
+  async fetchAllIssuers() {
+    try {
+      return await this.store.query('pki/issuer', { backend: this.secretMountPath.currentPath });
+    } catch (e) {
+      return e.httpStatus;
+    }
+  }
+
+  async fetchAllCertificates() {
```

Comment:
I don't think we need to fetch certificates here. The SearchSelect component handles querying models to populate the dropdown. This route just needs to query for issuers and roles to get those total counts for those two select cards, right?